### ISSUE TITLE
Sudo/Root call without fee

### ISSUE THREAD

**Issue by noahsalvadordenjo** [paritytech/substrate#7011]
Dear substrate,
I don't know if it should work this way, but when we use ```ensure_root``` and
call our extrinsic
```
await api.tx.sudo.sudo(api.tx.access.access('5CiKbPQtCW6X6oFpsLLJ8vSUEzqSYnUkWuoqkp41iDevMBK6', [1,2])).signAndSend(pair)
```
we have to pay some fee eventhough we have the following weigth annotation
```
#[weight = (100_000, DispatchClass::Normal, Pays::No)]
```
Is it possible for sudo calls to be free as well?

**Comment by thiolliere** [comment]
sudo should already be free when the extrinsic is successful (i.e. signed by sudo) as it returns Pays::No:
```rust
		fn sudo(origin, call: Box<<T as Trait>::Call>) -> DispatchResultWithPostInfo {
			// This is a public call, so we ensure that the origin is some signed account.
			let sender = ensure_signed(origin)?;
			ensure!(sender == Self::key(), Error::<T>::RequireSudo);

			let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());
			Self::deposit_event(RawEvent::Sudid(res.map(|_| ()).map_err(|e| e.error)));
			// Sudo user does not pay a fee.
			Ok(Pays::No.into())
		}
```

What is exactly your sudo extrinsic ?
also the current sudo does not pay fee only if the extrinsic is successful, that means that the signer must be able to pay the fee for the transaction to be valid (in case the transaction is not successful).

but indeed the following pattern
```
#[weight = (100_000, DispatchClass::Normal, Pays::No)]
```
shouldn't require the signer to be able to pay the fee (because no matter what he won't pay for it), is this your issue ?

**Comment by noahsalvadordenjo** [comment]
Yeah because of ```ensure_root``` instead of ```ensure_signed``` inside of the function body with the following annotation
```#[weight = (100_000, DispatchClass::Normal, Pays::No)]``` it charges a fee of +- 250000000

**Comment by shawntabrizi** [comment]
@noahsalvadordenjo this is solved in the latest versions of Substrate: https://github.com/paritytech/substrate/pull/6749

There is not much we can do about RC-4 I think, but you should get the expected behavior moving forward.

**Comment by thiolliere** [comment]
(note I closed because I couldn't reproduced on rc-6 and shawn says it was fixed, if you still have issue, feel free to reopen)

**Comment by noahsalvadordenjo** [comment]
couldn't reproduce as well
